### DISCUSSION TITLE
Three potential issues in jsoncons

### DISCUSSION THREAD

**Discussion by yimingming** [danielaparker/jsoncons#366] [Q&A]
Hello experts,
Based on the static coding analysis for jsoncons, it seems that the following issues need your double check.
1.   in  jsoncons/include/jsoncons/detail/write_number.hpp , 
     line 366:  char buffer[100];          it is not initialized.
     line 433:  char buffer[100];          it is not initialized.
2.  in  jsoncons/include/jsoncons/bigint.hpp,
     line 211:  The constructor allocates field data_ of jsoncons::basic_bigint<std::allocator<unsigned char> >::dynamic_storage but there is no destructor. Resource leak.
Could you please check if it has the potential risk. Thanks for your feedback in advance.

**Comment by danielaparker** [comment]
> 1. in  jsoncons/include/jsoncons/detail/write_number.hpp ,
>    line 366:  char buffer[100];          it is not initialized.
>    line 433:  char buffer[100];          it is not initialized.
> 2. in  jsoncons/include/jsoncons/bigint.hpp,
>    line 211:  The constructor allocates field data_ of jsoncons::basic_bigint<std::allocator >::dynamic_storage but there is no destructor. Resource leak.
>    Could you please check if it has the potential risk. Thanks for your feedback in advance.

Regarding (1), it's just a buffer of 100 chars, passed to another function that needs a temporary work space, initialization would serve no purpose, one initial value is as good as any other, they're never examined. We could initialize `buffer[0]` to `0`, if that would make the analyzer happy.

Regarding (2), there is no resource leak in a `bigint` object, all memory is allocated and deallocated, but there is an asymmetry, the memory is allocated in the constructor of `dynamic_storage` (which is a data member of `bigint`), and deallocated in the destructor of `bigint` (via a function call on the `dynamic_storage` member.) While that's not a risk, it's not good design either, we should change it, thanks for identifying this issue.

**Reply by yimingming** [reply]
Danielaparker,
Thanks for your quick feedback.
